[terra-core-docs] New Accessibility and Implementation guide for Terra Search Field

nikhitasharma opened the pull request:
<!-- 
PLEASE COMPLETE THESE STEPS BEFORE PUBLISHING THE PULL REQUEST:

*Before publishing*
1. Update the title as: [component] changes to component (ex. [terra-button] Added terra-button accessibility guide).
2. Fill out the fields below.
3. Assign yourself to the PR.
4. Add the appropriate package name labels.
5. Add your name to the CONTRIBUTORS.md file. Adding your name to the CONTRIBUTORS.md file signifies agreement to all rights and reservations provided by the License.
-->

### Summary
<!--- Summarize and explain the reasoning behind these code changes. What are the changes, and why are they necessary? -->

**What was changed:**
Added new Accessibility and Implementation guide for Terra Search Field


### Testing
<!-- Demonstrate that these changes are stable. How have these changes been verified? -->

This change was tested using:

- [ ] WDIO
- [ ] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [x] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:
<!-- Please include the appropriate "Required" & "Ready" labels (ex. "UX Review Required" and "UX Review Ready").  -->
- [x] UX review
- [x] Accessibility review
- [ ] Functional review

### Additional Details
<!-- List anything else that is relevant to this issue. Additional information will help us better understand your changes and speed up the review process. -->

**This PR resolves:**

UXPLATFORM-9618 <!-- Jira Number or issue Number. Please do not link to Jira. -->

---

Thank you for contributing to Terra.
@cerner/terra

chrismichalewicz commented:
@nikhitasharma  I haven't done a full review of it yet but from quickly glancing through it on the various examples can you call out which props were used to build out each example?  

Also added the tech writer review label and logged OHAIUAD-204813

nikhitasharma commented:
> @nikhitasharma I haven't done a full review of it yet but from quickly glancing through it on the various examples can you call out which props were used to build out each example?
> 
> Also added the tech writer review label and logged OHAIUAD-204813

I have added prop information which are being used to build each example.

eawww reviewed (commented):
To encourage best practices, could you add a groupName and isLabelVisible to the examples and make a special example for hidden label?

nikhitasharma commented on packages/terra-core-docs/src/terra-dev-site/doc/search-field/About.1.doc.mdx:
```diff
@@ -1,35 +1,90 @@
+
+List of keyboard interactions supported by Search Field
+
+|Key interactions|Description|
+|---|---|
+|Tab           | Moves focus into the search field. If already on the search text box with text entered, tab moves focus to the clear text button. If already on clear text button, tab moves followed by focus to the search button.|
+|Shift + Tab    | Moves focus to the previous focusable element in the tab sequence.|
+|Space| When on clear button, clears the current search text. When on search button, triggers the search callback.|
+|Enter| When on clear button, clears the current search text. When on search button, triggers the search callback.|
```
@chrismichalewicz  Do you agree that we need not have this. Making sure this is okay since the template you shared had the Keyboard interactions section.

mjpalazzo replied:
My preference is to include all keyboard interactions supported by the component. This is what ARIA APG site does, and it helps reinforce to consumers who may not be familiar with accessibility how things are supposed to work.

chrismichalewicz replied:
@nikhitasharma  I agree with @mjpalazzo  I think it is good to include all support keyboard interactions even if we think they are common.  Even so we are making an assumption that all users know those commands and that might not be the case.

chrismichalewicz reviewed (approved):
@nikhitasharma Thank you for updating the examples with the props used. I think this page looks good from a functional perspective.

nikhitasharma commented:
> To encourage best practices, could you add a groupName and isLabelVisible to the examples and make a special example for hidden label?

I have added an example with no label.

mjpalazzo commented:
Great work @nikhitasharma! Here are some minor observations:
Feature section, bullet 10, management is misspelled in this sentence, "Mangement of invalid search text states."
Example section, the description column might add more value if you referenced the prop that the example demonstrates.
Accessibility section, Assistive Technology Support sub-section, we should provide guidance about using the placeholder text property (for why it has potential accessibility issues, see https://www.w3.org/WAI/tutorials/forms/instructions/#placeholder-text) @eawww - Terra supports placeholder text for many input components. Would it be possible to create some standard guidance that can used for each relevant component?

eawww commented:
> Great work @nikhitasharma! Here are some minor observations: Feature section, bullet 10, management is misspelled in this sentence, "Mangement of invalid search text states." Example section, the description column might add more value if you referenced the prop that the example demonstrates. Accessibility section, Assistive Technology Support sub-section, we should provide guidance about using the placeholder text property (for why it has potential accessibility issues, see https://www.w3.org/WAI/tutorials/forms/instructions/#placeholder-text) @eawww - Terra supports placeholder text for many input components. Would it be possible to create some standard guidance that can used for each relevant component?

Design Standards official blurb on Placeholder Text might be enough:
https://wiki.cerner.com/display/UserExperience/Input+Fields

Placeholder text
- Placeholder text is italicized text that lives within the input field but disappears when the user starts typing inside the form field. 
- Placeholder text should not be relied on in order to prevent errors from occurring later, instead use masked entry, helper text, or a combination of both.
- Placeholder text is only appropriate when the information it conveys is not vital to the completion of the form, as such, placeholder text is reserved for specific components to provide additional details as to the type of input field:
  - Dropdown
  - Search

nikhitasharma commented:
> @mjpalazzo @nikhitasharma
> 
> > Accessibility section, Assistive Technology Support sub-section, we should provide guidance about using the placeholder text property (for why it has potential accessibility issues, see https://www.w3.org/WAI/tutorials/forms/instructions/#placeholder-text) @eawww - Terra supports placeholder text for many input components. Would it be possible to create some standard guidance that can used for each relevant component?
> 
> Design Standards official blurb on Placeholder Text might be enough: https://wiki.cerner.com/display/UserExperience/Input+Fields
> 
> Placeholder text
> 
> * Placeholder text is italicized text that lives within the input field but disappears when the user starts typing inside the form field.
> * Placeholder text should not be relied on in order to prevent errors from occurring later, instead use masked entry, helper text, or a combination of both.
> * Placeholder text is only appropriate when the information it conveys is not vital to the completion of the form, as such, placeholder text is reserved for specific components to provide additional details as to the type of input field:
>   
>   * Dropdown
>   * Search

Added this and addressed other comments as well.

mjpalazzo commented:
Looks good, @nikhitasharma! Thanks for making the revisions.
